src/math/differences.hpp

```diff
@@ -17,6 +17,12 @@ constexpr bool equals(const Eigen::MatrixBase<DerivedA> &A,
   return A.isApprox(B, tolerance);
 }
 
+/// Compares two doubles for equality up to tolerance
+constexpr inline bool equals(double a, double b, double tolerance = NUMERICAL_ZERO_DIFFERENCE)
+{
+  return std::abs(a - b) < tolerance;
+}
+
```

Comment:
Why do we need this additional function? Doesn't the function just below here do the job?

Reply:
You are right! Thanks for pointing that out!